guiEditor/src/diagram/workbench.tsx

```diff
@@ -90,6 +95,27 @@ export class WorkbenchComponent extends React.Component<IWorkbenchComponentProps
         return this._selectedGuiNodes;
     }
 
+    // given a depth and a control gets the parent up the tree depth times
+    private _getParentWithDepth(control: Control, depth: number) {
+        let parent = control;
+        for (let i = 0; i < depth; ++i) {
+            if (parent.parent)
+                parent = parent.parent;
+        }
+        return parent;
+    }
+
+    //gets the higher parent of a given control.
+    private _getMaxParent(control: Control) {
+        let parent = control;
+        this._selectionDepth = 0;

```

Comment:
This is strange that _getMaxParent is changing selectionDepth when _getParentWithDepth does not ?

It looks like either both should and in this case I wonder if the should not also select and be renamed, or both should not ?